### PR TITLE
runProtocol ProtocolRealPBFT:  derive from genesis

### PR DESCRIPTION
**Pull request by deepfire**

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs [hunk 1]
```diff
@@ -79,10 +78,10 @@ data Protocol blk where
 
   -- | Run PBFT against the real ledger
   ProtocolRealPBFT
-    :: Cardano.Genesis.Config
+    :: Genesis.Config
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs]: Maybe we should switch to record syntax?

Comment (deepfire) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs]: I was thinking of it, but dared not to suggest, assuming important reasons for not doing so : -)

Comment (dcoutts) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs]: It would make the `runProtocol` pattern match below a lot clearer.

Comment (deepfire) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs]: Are we not afraid of partial field accessors?

We had a discussion about them in a similar situation: https://github.com/input-output-hk/cardano-node/pull/147#discussion_r317308688

Comment (mrBliss) [reply ouroboros-consensus/src/Ouroboros/Consensus/Protocol.hs]: In this case the fields make the code more understandable. IMO, one shouldn't use field accessor functions anyway (https://github.com/ghc-proposals/ghc-proposals/pull/160) 🙂 (Maybe with newtypes, alright).

**Pull request merged**